test/app-tap/httpd.py

```diff
@@ -1,4 +1,4 @@
-#!/usr/bin/env python2
+#!/usr/bin/env python3
```

Comment:
The `app-tap/http_client.test.lua` remains the only test that is not Python version agnostic. We can change it to run `httpd.py` using the same Python interpreter as one that run test-run (but keep ability to run it without test-run):

```diff
diff --git a/test/app-tap/http_client.skipcond b/test/app-tap/http_client.skipcond
index 9c1ca22b1..1047177d4 100644
--- a/test/app-tap/http_client.skipcond
+++ b/test/app-tap/http_client.skipcond
@@ -1,7 +1,14 @@
+import os
+import sys
 import platform
 
 # Disabled on OpenBSD due to fail #XXXX.
 if platform.system() == 'OpenBSD':
     self.skip = 1
 
+# We use gevent in httpd.py as well as in test-run. Many
+# interpreter versions may be installed on a system, but
+# one that runs test-run surely has the gevent library.
+os.environ['PYTHON_EXECUTABLE'] = sys.executable
+
 # vim: set ft=python:
diff --git a/test/app-tap/http_client.test.lua b/test/app-tap/http_client.test.lua
index aafad0004..3029ee4b6 100755
--- a/test/app-tap/http_client.test.lua
+++ b/test/app-tap/http_client.test.lua
@@ -36,8 +36,11 @@ local function start_server(test, sock_family, sock_addr)
     else
         error(string.format('invalid socket family: %s', sock_family))
     end
-    local cmd = string.format("%s/test/app-tap/httpd.py %s",
-                              TARANTOOL_SRC_DIR, arg)
+    -- PYTHON_EXECUTABLE is set in http_client.skipcond.
+    local python_executable = os.getenv('PYTHON_EXECUTABLE') or ''
+    local cmd_prefix = (python_executable .. ' '):lstrip()
+    local cmd = string.format("%s%s/test/app-tap/httpd.py %s",
+                              cmd_prefix, TARANTOOL_SRC_DIR, arg)
     local server = io.popen(cmd)
     test:is(server:read("*l"), "heartbeat", "server started")
     test:diag("trying to connect to %s", url)
diff --git a/test/app-tap/httpd.py b/test/app-tap/httpd.py
index 60084bdbf..a3dc1a3a8 100755
--- a/test/app-tap/httpd.py
+++ b/test/app-tap/httpd.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python2
+#!/usr/bin/env python
 
 import sys
 from gevent.pywsgi import WSGIServer
```

(It would be better to set the environment variable in test-run, but the change will not be so local this way.)

It may be negligible or important depending of how we'll update tests, test-run and testing infrastructure.

Reply:
replaced my own patch with yours